yarn-project/end-to-end/src/e2e_token_contract.test.ts

```diff
@@ -496,6 +496,31 @@ describe('e2e_token_contract', () => {
           );
         });
 
+        it('transfer on behalf of other fails if approval is removed', async () => {
+          const balance0 = await asset.methods.balance_of_private(accounts[0].address).view();
+          const amount = balance0 / 2n;
+          const nonce = Fr.random();
+          expect(amount).toBeGreaterThan(0n);
+
+          // docs:start:remove_private_auth_witness
+          // create auth witness
+          const action = asset
+            .withWallet(wallets[1])
+            .methods.transfer(accounts[0].address, accounts[1].address, amount, nonce);
+          const messageHash = await computeAuthWitMessageHash(accounts[1].address, action.request());
+          const witness = await wallets[0].createAuthWitness(messageHash);
+          await wallets[1].addAuthWitness(witness);
+
+          // wallets[0] want to undo their authorization action. So they tell wallets[1] to do so:
+          await wallets[1].removePrivateAuthWitness(messageHash);
```

Comment:
I think having two wallets being used here, especially since this is being used in the docs is more confusing than clarifying 

If it needs to stay, then maybe renaming them instead of saying wallets[0] / wallets[1] give them names, such as alice / bob or something